### PR TITLE
relay: begin with the latest binlog file if relay position is not set

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
### What problem does this PR solve? 
relay begin with the latest binlog file if relay position is not set

### Check List 

Tests 

 - Manual test (don't set relay log's config, and check the binlog file's start name under relay log's dir)

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: relay/relay.go [hunk 1]
```diff
@@ -498,14 +498,21 @@ func (r *Relay) reSetupMeta() error {
 		return err
 	}
 
+	lastPos, _, err := utils.GetMasterStatus(r.db, r.cfg.Flavor)
```
Comment (csuzhangxc) [inline relay/relay.go]: do not support fetching binlog from the newest GTID sets?

Is any better name than `lastPos`? like `latestPos` or `newest`?

Comment (WangXiangUSTC) [reply relay/relay.go]: my mistake, fixed in https://github.com/pingcap/dm/pull/304/commits/36603239528f65c8a27f7cd1c9660cc920d2a65b

**Comment by csuzhangxc** [comment]
can you and some test cases to cover this logic? like mock `r.db` for `SHOW MASTER STATUS`?

**Comment by csuzhangxc** [comment]
> can you and some test cases to cover this logic? like mock `r.db` for `SHOW MASTER STATUS`?

Oh, maybe cases in _realy/meta_test.go_ are enough.

File: relay/relay.go [hunk 2]
```diff
@@ -498,14 +498,21 @@ func (r *Relay) reSetupMeta() error {
 		return err
 	}
 
+	latestPos, latestGTID, err := utils.GetMasterStatus(r.db, r.cfg.Flavor)
```
Comment (csuzhangxc) [inline relay/relay.go]: should we always call `GetMasterStatus` or only when not pos/GTIDs are given?

Comment (WangXiangUSTC) [reply relay/relay.go]: update in https://github.com/pingcap/dm/pull/304/commits/cd77953171a8b31e7a69f0fcb14c8fec8e0be5cd

**Comment by WangXiangUSTC** [comment]
> > can you and some test cases to cover this logic? like mock `r.db` for `SHOW MASTER STATUS`?
> 
> Oh, maybe cases in _realy/meta_test.go_ are enough.
> or add cases with both `binlogName` and `latestBinlogName` are not empty.

update in https://github.com/pingcap/dm/pull/304/commits/0909418297c854c33e853c7631dd3048df699c2e

**Comment by csuzhangxc** [comment]
@amyangfei PTAL

File: relay/meta.go [hunk 3]
```diff
@@ -165,30 +165,30 @@ func (lm *LocalMeta) AdjustWithStartPos(binlogName string, binlogGTID string, en
...
 		gset, err = gtid.ParserGTID(lm.flavor, binlogGTID)
 		if err != nil {
 			return false, terror.Annotatef(err, "relay-binlog-gtid %s", binlogGTID)
 		}
-	}
-
-	// verified, update them
-	if enableGTID {
-		lm.BinLogName = minCheckpoint.Name
+		lm.BinLogName = ""
```
Comment (amyangfei) [inline relay/meta.go]: If gtid is enabled in the relay module, the `BinlogName` is useless when pulling relay log, but the relay module still saves a `BinlogName` based on the binlog position each time when flushing relay meta. Shall we unify the usage of `BinlogName` in relay meta?

Comment (csuzhangxc) [reply relay/meta.go]: if we get GTID from the upstream master server, we can also get BinlogName at the same time. but if we get GTID from the startup config file, BinlogName is often empty.

But we may still assign BinlogName to meta.

Comment (WangXiangUSTC) [reply relay/meta.go]: so, should we set `BinlogName` here  when enable gtid? @csuzhangxc @amyangfei

Comment (amyangfei) [reply relay/meta.go]: I prefer to set `BinlogName`

Comment (WangXiangUSTC) [reply relay/meta.go]: update in https://github.com/pingcap/dm/pull/304/commits/e6d94862de9f2676332fc2f0a4f36a261b6e8b35, do you think it's ok @csuzhangxc @amyangfei

**Pull request merged**